service/history/workflow/mutable_state_impl.go

```diff
@@ -640,6 +642,21 @@ func (ms *MutableStateImpl) chasmEnabled() bool {
 	return !isNoop
 }
 
+// getChasmWorkflowComponent gets the root workflow component from the CHASM tree.
+// Returns both the workflow component and the CHASM mutable context.
+func (ms *MutableStateImpl) getChasmWorkflowComponent(ctx context.Context) (*chasmworkflow.Workflow, chasm.MutableContext, error) {
```

Comment:
Please don't use `get` for getters in Go.

```suggestion
func (ms *MutableStateImpl) chasmWorkflowComponent(ctx context.Context) (*chasmworkflow.Workflow, chasm.MutableContext, error) {
```